src/Client/Settings.java

```diff
@@ -2707,6 +2646,32 @@ public static Properties initSettings() {
     return null;
   }
 
+  /**
+   * Saves just the "current_profile" value to the config.ini, used for saving presets
+   *
+   * @param profile
+   */
+  static void saveProfile(String profile) {
+    if (!successfullyInitted) {
+      Logger.Warn(
+          "Prevented erroneous save, please report this along with the RSC+ log file, set to debug logging mode");
+      return;
+    }
+
+    try {
+      Properties props = loadProps();
+
+      props.setProperty("current_profile", profile);
+
+      FileOutputStream out = new FileOutputStream(Dir.JAR + "/config.ini");
+      props.store(out, "---rscplus config---");
+      out.close();
+    } catch (Exception e) {
+      e.printStackTrace();
+      Logger.Error("Unable to save settings");
+    }
+  }
+
```

Comment:
Don't understand what this is for.

Reply:
It writes only the "current_profile" setting to config.ini. It's needed for the `Save Current Preset to settings` button - if you look at its logic, it saves the currently-chosen profile to config.ini, which includes its own value for `current_profile` - this needs to then be re-saved as `custom`, but it felt like overkill to re-save the ENTIRE file, so I added this.

Reply:
To be honest I'm not sure why `current_profile` is saved in config.ini. It looks like it might have been my original intention was in fact that you could save your selected profile and load into it on startup:

L2591 Settings.java: `currentProfile = getPropString(props, "current_profile", "custom");`, in the `initSettings()` function. But this doesn't actually ever happen, maybe because of the lines you added just recently:

```
    if (!currentProfile.equals("custom")) {
      currentProfile = "custom";
    }
```

If there's no reason why `current_profile` would ever need to be saved and loaded on launch, it shouldn't be written to config.ini, so this function would not be necessary.

Reply:
Thanks! Now that you mention it, I actually had a feeling it might not be needed a couple days ago and totally forgot to check it. Will confirm before the next review revision ^_^

Reply:
^_^

Reply:
I can't think of any reason for it, this seems fine, thanks!